handler/proxy.go

```diff
@@ -330,6 +348,26 @@ func (p *Proxy) Director(req *http.Request) error {
 	req.URL.Scheme = originURL.Scheme
 	req.Host = originURL.Host
 
+	query := req.URL.Query()
```

Comment:
This block should be a) extracted to own method as already mentioned and b) splits the "read / logic flow" since the if condition below is related to set the request host above.